src/test/java/com/google/javascript/clutz/partial/global_generic_type.js

```diff
@@ -0,0 +1,8 @@
+goog.provide('global.generic.type');
+goog.provide('global.non.generic.type');
+
```

Comment:
There is one more weird case where closure returns type arguments as different type than NoType. So add a test for:

/** @type {!SomeType<!Map<string, string>>  */

Reply:
Done, took major surgery, take a look at the diff.